python/ray/autoscaler/resource_demand_scheduler.py

```diff
@@ -64,42 +65,51 @@ def get_nodes_to_launch(self, nodes: List[NodeID],
             return []
 
         node_resources, node_type_counts = self.calculate_node_resources(
-            nodes, pending_nodes)
+            nodes, pending_nodes, usage_by_ip)
+
         logger.info("Cluster resources: {}".format(node_resources))
         logger.info("Node counts: {}".format(node_type_counts))
 
         unfulfilled = get_bin_pack_residual(node_resources, resource_demands)
         logger.info("Resource demands: {}".format(resource_demands))
         logger.info("Unfulfilled demands: {}".format(unfulfilled))
 
-        nodes = get_nodes_for(self.node_types, node_type_counts,
-                              self.max_workers - len(nodes), unfulfilled)
+        nodes = get_nodes_for(
+            self.node_types, node_type_counts,
+            self.max_workers - len(nodes) - len(pending_nodes), unfulfilled)
```

Comment:
incidental bug fix

Reply:
Can we add a unit test for this one too?